src/interface/storage.thrift

```diff
@@ -203,6 +205,40 @@ struct CatchUpDataReq {
     3: common.HostAddr     target,
 }
 
+struct UpdateResponse {
+    1: required ResponseCommon result,
+    2: optional common.Schema schema,   // return column related props schema
+    3: optional binary data,            // return column related props value
+    4: optional bool upsert = false,    // it's true when need to be inserted by UPSERT
+}
+
+struct UpdateItem {
+    1: required binary name,    // TagName or EdgeType
+    2: required binary field,   // property name
+    3: required binary value,   // expression which is encoded
+}
+
+struct UpdateVertexRequest {
+    1: common.GraphSpaceID space_id,
+    2: common.VertexID vertex_id,
+    3: common.PartitionID part_id,
+    4: binary filter,
+    5: list<UpdateItem> update_items,
+    6: list<PropDef> return_columns,
+    7: bool insertable,
+    8: common.EdgeType edge_type,
+}
+
+struct UpdateEdgeRequest {
+    1: common.GraphSpaceID space_id,
+    2: EdgeKey edge_key,
+    3: common.PartitionID part_id,
+    4: binary filter,
+    5: list<UpdateItem> update_items,
+    6: list<PropDef> return_columns,
+    7: bool insertable,
+    8: common.EdgeType edge_type,
```

Comment:
We already have edge_key,  why we need edge_type here.

Reply:
ditto!